### PR TITLE
store/helper, infoschema: add table info for region info mem tables

### PR DESCRIPTION
**Pull request by Luffbee**
### What problem does this PR solve? 

It's hard to get information about regions of one table or some specific tables (eg. regions distribution of one table: for each store, how many region peers of table A are there?) 

### What is changed and how it works?

Add table info for region info tables. Now they are like following:

```
mysql> select * from TIKV_REGION_PEERS limit 10;
+-----------+----------+---------+------------+----------+----------+------------+---------+----------+------------+-----------+--------+--------------+
| REGION_ID | TABLE_ID | DB_NAME | TABLE_NAME | IS_INDEX | INDEX_ID | INDEX_NAME | PEER_ID | STORE_ID | IS_LEARNER | IS_LEADER | STATUS | DOWN_SECONDS |
+-----------+----------+---------+------------+----------+----------+------------+---------+----------+------------+-----------+--------+--------------+
|        18 |       17 | mysql   | help_topic |        1 |        1 | name       |      19 |        1 |          0 |         0 | NORMAL |         NULL |
|        18 |       17 | mysql   | help_topic |        1 |        1 | name       |    1502 |      553 |          0 |         1 | NORMAL |         NULL |
|        18 |       17 | mysql   | help_topic |        1 |        1 | name       |    1725 |      296 |          0 |         0 | NORMAL |         NULL |
|        18 |       17 | mysql   | help_topic |        0 |     NULL | NULL       |      19 |        1 |          0 |         0 | NORMAL |         NULL |
|        18 |       17 | mysql   | help_topic |        0 |     NULL | NULL       |    1502 |      553 |          0 |         1 | NORMAL |         NULL |
|        18 |       17 | mysql   | help_topic |        0 |     NULL | NULL       |    1725 |      296 |          0 |         0 | NORMAL |         NULL |
|        20 |       19 | mysql   | stats_meta |        1 |        1 | idx_ver    |      21 |        1 |          0 |         0 | NORMAL |         NULL |
|        20 |       19 | mysql   | stats_meta |        1 |        1 | idx_ver    |    1503 |      553 |          0 |         0 | NORMAL |         NULL |
|        20 |       19 | mysql   | stats_meta |        1 |        1 | idx_ver    |    1548 |      296 |          0 |         1 | NORMAL |         NULL |
|        20 |       19 | mysql   | stats_meta |        1 |        2 | tbl        |      21 |        1 |          0 |         0 | NORMAL |         NULL |
+-----------+----------+---------+------------+----------+----------+------------+---------+----------+------------+-----------+--------+--------------+
10 rows in set (0.00 sec)

mysql> select * from TIKV_REGION_STATUS limit 10;
+-----------+--------------------------------------------------------+--------------------------------------------------------+---------------------+--------------------+----------------------------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+
| REGION_ID | START_KEY                                              | END_KEY                                                | TABLE_ID            | DB_NAME            | TABLE_NAME                       | IS_INDEX | INDEX_ID | INDEX_NAME | EPOCH_CONF_VER | EPOCH_VERSION | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE | APPROXIMATE_KEYS |
+-----------+--------------------------------------------------------+--------------------------------------------------------+---------------------+--------------------+----------------------------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+
|      4145 | 7480000000000000FF4B5F72D400000000FF0000000000000000FA | 7480000000000000FF4B5F72D800000000FF0000000000000000FA |                  75 | test               | tttt                             |        0 |     NULL | NULL       |              5 |           835 |             0 |          0 |                1 |                0 |
|      4189 | 7480000000000000FF4B5F698000000000FF0000010000000000FA | 7480000000000000FF4B5F698000000000FF0000020000000000FA |                  75 | test               | tttt                             |        1 |        1 | idx1       |              5 |           816 |             0 |          0 |                1 |                0 |
|         8 | 7480000000000000FF0700000000000000F8                   | 7480000000000000FF0900000000000000F8                   |                   7 | mysql              | db                               |        1 |        1 | PRIMARY    |              5 |             4 |             0 |          0 |                1 |             4283 |
|         8 | 7480000000000000FF0700000000000000F8                   | 7480000000000000FF0900000000000000F8                   |                   7 | mysql              | db                               |        0 |     NULL | NULL       |              5 |             4 |             0 |          0 |                1 |             4283 |
|      4049 | 7480000000000000FF4B5F72FC00000000FF0000000000000000FA |                                                        |                  75 | test               | tttt                             |        0 |     NULL | NULL       |              5 |           844 |             0 |          0 |                1 |                0 |
|      4049 | 7480000000000000FF4B5F72FC00000000FF0000000000000000FA |                                                        | 9223372036854775014 | PERFORMANCE_SCHEMA | events_stages_history_long       |        0 |     NULL | NULL       |              5 |           844 |             0 |          0 |                1 |                0 |
|      4049 | 7480000000000000FF4B5F72FC00000000FF0000000000000000FA |                                                        | 9223372036854775027 | PERFORMANCE_SCHEMA | events_stages_history            |        0 |     NULL | NULL       |              5 |           844 |             0 |          0 |                1 |                0 |
|      4049 | 7480000000000000FF4B5F72FC00000000FF0000000000000000FA |                                                        | 9223372036854775040 | PERFORMANCE_SCHEMA | events_stages_current            |        0 |     NULL | NULL       |              5 |           844 |             0 |          0 |                1 |                0 |
|      4049 | 7480000000000000FF4B5F72FC00000000FF0000000000000000FA |                                                        | 9223372036854775065 | PERFORMANCE_SCHEMA | events_transactions_history_long |        0 |     NULL | NULL       |              5 |           844 |             0 |          0 |                1 |                0 |
|      4049 | 7480000000000000FF4B5F72FC00000000FF0000000000000000FA |                                                        | 9223372036854775090 | PERFORMANCE_SCHEMA | events_transactions_history      |        0 |     NULL | NULL       |              5 |           844 |             0 |          0 |                1 |                0 |
+-----------+--------------------------------------------------------+--------------------------------------------------------+---------------------+--------------------+----------------------------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+
10 rows in set (0.00 sec)
```

For the regions distribution problem, we can use the following query:

``` sql
SELECT COUNT(region_id),
       table_name,
       index_name,
       store_id
FROM TIKV_REGION_PEERS
WHERE db_name='test'
  AND table_name='tttt'
GROUP BY table_name,
         index_name,
         store_id
ORDER BY table_name DESC, index_name
LIMIT 15;
```

which will give something like this: 

```
+------------------+------------+------------+----------+
| COUNT(region_id) | table_name | index_name | store_id |
+------------------+------------+------------+----------+
|               32 | tttt       | NULL       |      553 |
|               32 | tttt       | NULL       |      296 |
|               32 | tttt       | NULL       |        1 |
|                1 | tttt       | idx1       |      296 |
|                1 | tttt       | idx1       |      553 |
|                1 | tttt       | idx1       |        1 |
|                1 | tttt       | idx2       |        1 |
|                1 | tttt       | idx2       |      296 |
|                1 | tttt       | idx2       |      553 |
|                1 | tttt       | idx3       |      553 |
|                1 | tttt       | idx3       |      296 |
|                1 | tttt       | idx3       |        1 |
+------------------+------------+------------+----------+
```

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Related changes

 - Need to be included in the release note

### CODE REVIEW

File: store/helper/helper.go [hunk 1]
```diff
@@ -399,36 +412,208 @@ type RegionsInfo struct {
...
+	regions := []*RegionInfo{}
+	for i := 0; i < len(regionsInfo.Regions); i++ {
+		tableInfos[regionsInfo.Regions[i].ID] = []TableInfo{}
+		regions = append(regions, &regionsInfo.Regions[i])
+	}
+
+	tables := []tableInfoWithKeyRange{}
+	for _, db := range schemas {
+		for _, table := range db.Tables {
+			tables = append(tables, newTableWithKeyRange(db, table))
```
Comment (nolouch) [inline store/helper/helper.go]: Can we reuse the function `FindTableIndexOfRegion`?

Comment (Luffbee) [reply store/helper/helper.go]: That function may cause performance issues. It uses `O(M)` time to find table infos for one region, where `M` is the number of tables and indices. So, it takes `O(MN)` time to find table infos for `N` regions. My implementation only need `O(M log(M) + N log(N) + kN)` time, where `k` is the average number of tables and indices in one region(which is usually very small). For hot regions (`N` is not very large), maybe `O(MN)` is not a problem, but this PR is working on all regions.
What's more, `FindTableIndexOfRegion`  assumes there is only one table or index in one region, which is not true (I'm not sure is this a bug).

**Comment by Luffbee** [comment]
@winoros

File: store/helper/helper.go (hunk 1 above)
Comment (winoros) [reply store/helper/helper.go]: `FindTableIndexOfRegion` is copied from the old codes in package `server`.

Comment (Luffbee) [reply store/helper/helper.go]: > `FindTableIndexOfRegion` is copied from the old codes in package `server`. If the assumption is break, you can change the implementation of it.

It seems that will change a lot of old codes. I think that should be another PR.

**Pull request merged**